source/std.tex

```diff
@@ -72,6 +72,7 @@
 %%--------------------------------------------------
 %% add special hyphenation rules
 \hyphenation{tem-plate ex-am-ple in-put-it-er-a-tor name-space name-spaces}
+\hyphenation{non-ze-ro}
```

Comment:
Indeed. You can't weigh the hints, so I'd much rather only give the good hint, and if there's really a line that needs hyphenation at the second point, I'd provide that as a manual hint on a one-off basis.